micro-rdk/src/esp32/tls.rs

```diff
@@ -176,11 +174,13 @@ impl Esp32TLS {
         Self {
             alpn_ptr,
             tls_cfg: Either::Right(tls_cfg_client),
+            cfg: None,
         }
     }
     /// Creates a TLS object ready to accept connection or connect to a server
-    pub fn new_server(cfg: &Esp32TLSServerConfig) -> Self {
+    pub fn new_server(cfg: Esp32TLSServerConfig) -> Self {
         let mut alpn_ptr: Vec<_> = vec![ALPN_PROTOCOLS.as_ptr() as *const i8, std::ptr::null()];
+        let cfg = Box::new(cfg);
```

Comment:
See below. There is probably a better way to do this, maybe with `Pin`ing?

Reply:
I think you're right about `Pin`ing. With Box there's no guarantee that the underlying data doesn't get manipulated/moved-out/invalidated/etc. As long as it's not a `PhantomPin` it should be cleaned up once dropped out of scope.